src/components/NewExperiment.vue

```diff
@@ -1198,22 +1274,91 @@ export default Vue.extend({
           });
       });
 
-      // parsedRows = [[["name", "a"], ["measurement", 5], ["uncertainty", null]]]
-      parsedRows.forEach((rowPairs, rowIx) => {
-        if (!table.items[rowOffset + rowIx]) {
-          // Create excess rows.
-          table.items.push({ temporaryId: uuidv4() });
-        }
+      let itemType;
+      let items;
+      if (table.name === "Samples") {
+        itemType = "condition";
+        items = this.tables.conditions.items.filter(
+          condition => condition.name
+        );
+      } else {
+        itemType = "sample";
+        items = this.tables.samples.items.filter(sample => sample.name);
+      }
+
+      // Ask which condition/sample pasted data belongs to
+      const selection =
+        table.name === "Conditions"
+          ? Promise.resolve(true)
+          : this.$promisedDialog(SelectDialog, {
+              itemType: itemType,
+              items: items
+            });
 
-        rowPairs.forEach(([property, value]) => {
-          Vue.set(table.items[rowOffset + rowIx], property, value);
+      selection.then(selectedItem => {
+        // parsedRows = [[["name", "a"], ["measurement", 5], ["uncertainty", null]]]
+        parsedRows.forEach((rowPairs, rowIx) => {
+          if (!table.items[rowOffset + rowIx]) {
+            // Create excess rows.
+            table.items.push({ temporaryId: uuidv4() });
+          }
+          // Autoselect condition/sample that was chosen by user
+          if (table.name !== "Conditions") {
+            Vue.set(table.items[rowOffset + rowIx], itemType, selectedItem);
+          }
+          setTimeout(() => {
+            this.$refs.newExperimentForm.validate();
+          }, 0);
+          rowPairs.forEach(([property, value]) => {
+            // Autoselect the exact match for strain and medium
+            if (table.name === "Conditions") {
+              if (property === "strain") {
+                value =
+                  this.availableStrains.find(
+                    strain => strain.name === value._pastedText
+                  ) || value;
+              } else if (property === "medium") {
+                value =
+                  this.availableMedia.find(
+                    medium => medium.name === value._pastedText
+                  ) || value;
+              }
```

Comment:
This can be moved into parsePasted
```ts
parsePasted: {
  name: str => str,
  strain: (str, { availableStrains }) => {
    const match = availableStrains.find(({ name }) => name === str);
    return match || { _pastedText: str };
  },
  medium: (str, { availableMedia }) => {
    const match = availableMedia.find(({ name }) => name === str);
    return match || { _pastedText: str };
  }
},
```

You just need to append this to "Extra parameters for parsePasted:"
```
  availableStrains: this.availableStrains,
  availableMedia: this.availableMedia
```

Reply:
Thank you! That's perfect